bucket.go

```diff
@@ -141,9 +141,9 @@ func (ve ViewError) Error() string {
 	return fmt.Sprintf("Node: %v, reason: %v", ve.From, ve.Reason)
 }
 
-type UpdateFunc func(current []byte) (updated []byte, err error)
+type UpdateFunc func(current []byte) (updated []byte, expiry *uint32, err error)
 
-type WriteUpdateFunc func(current []byte) (updated []byte, opt WriteOptions, err error)
+type WriteUpdateFunc func(current []byte) (updated []byte, opt WriteOptions, expiry *uint32, err error)
```

Comment:
What is the rationale behind this returning an `*uint32` vs a `uint32`?   Is that to distinguish between expiry and no expiry?  I was under the impression that `0` meant no expiry in the gocb usage, just wondering if we should stick to that convention for consistency.

Reply:
This was to be able to identify whether the Sync Function had explicitly set an expiry of zero (which would override any other expiry settings), or hadn't set the expiry.